docs/reference/indices/diskusage.asciidoc

```diff
@@ -0,0 +1,194 @@
+[[indices-disk-usage]]
+=== Analyze index disk usage API
+++++
+<titleabbrev>Analyze index disk usage</titleabbrev>
+++++
+
+experimental[]
+
+Analyzes the disk usage of each field of an index. This API
+might not support indices created in previous {es} versions.
+The result of a small index can be inaccurate as some parts
+of an index might not be analyzed by the API. It's recommended
+to execute this API with a single index (or the latest backing
+index of a data streams) as the API consumes resources significantly.
+
+[source,console]
+--------------------------------------------------
+POST /my-index-000001/_disk_usage?run_expensive_tasks=true
+--------------------------------------------------
+// TEST[setup:messages]
+
+[[analyze-index-disk-usage-api-request]]
+==== {api-request-title}
+
+`POST /<index>/_disk_usage`
+
+[[analyze-index-disk-usage-api-request-prereqs]]
+==== {api-prereq-title}
+
+* If the {es} {security-features} are enabled, you must have the `manage`
+<<privileges-list-indices,index privilege>> for the target index or index alias.
+
+[[analyze-index-disk-usage-api-path-params]]
+==== {api-path-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index]
+
+[[analyze-index-disk-usage-api-query-params]]
+==== {api-query-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
++
+Defaults to `true`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
++
+Defaults to `open`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=wait_for_active_shards]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
+
+`run_expensive_tasks`::
+(Optional, Boolean)
+Since analyzing the per field disk usage of indices is expensive and resource-intensive,
+the parameter `run_expensive_tasks` must be set to `true` in order for the task to be
+performed.
+Defaults to `false`.
+
+`flush`::
+(Optional, Boolean)
+If `true`, the target index will be flushed before the analyzing process starts.
+If `false`, then uncommitted data might not be included in the result.
+Defaults to `true`.
```

Comment:
I sorted these in alphabetical order. I also made some minor edits to the `flush` and `run_expensive_tasks` params.
```suggestion
include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
+
Defaults to `true`.

include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
+
Defaults to `open`.

`flush`::
(Optional, Boolean) If `true`, the API performs a flush before analysis. If
`false`, the response may not include uncommitted data. Defaults to `true`.

include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]

include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=timeoutparms]

`run_expensive_tasks`::
(Required, Boolean) Analyzing field disk usage is resource-intensive. To use the
API, this parameter must `true`. Defaults to `false`.

include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=wait_for_active_shards]
```